### PR TITLE
fix(debug): add config, parent_config to checkpoint debug events, add state to debug tasks

### PR DESCRIPTION
**Pull request by dqbd**

### CODE REVIEW

File: libs/langgraph/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "build:internal": "yarn clean && yarn lc_build --create-entrypoints --pre --tree-shaking",
     "clean": "rm -rf dist/ dist-cjs/ .turbo/",
     "lint:eslint": "NODE_OPTIONS=--max-old-space-size=4096 eslint --cache --ext .ts,.js src/",
-    "lint:dpdm": "dpdm --exit-code circular:1 --no-warning --no-tree src/*.ts src/**/*.ts",
+    "lint:dpdm": "dpdm --exit-code circular:1 --no-warning --no-tree --transform src/*.ts src/**/*.ts",
```
Comment (dqbd) [inline libs/langgraph/package.json:20]: Circular type imports are I think fine.

File: libs/langgraph/src/pregel/debug.ts [hunk 2]
```diff
@@ -186,12 +188,24 @@ export function* mapDebugCheckpoint<
...
+    taskStates[task.id] = {
+      configurable: {
+        thread_id: config.configurable?.thread_id,
+        checkpoint_ns: taskNs,
+      },
+    };
   }
 
-  const ts = getCurrentUTC().toISOString();
+  const ts = new Date().toISOString();
```
Comment (dqbd) [inline libs/langgraph/src/pregel/debug.ts:208]: This is the only place which tries to return UTC date, whereas rest of the app uses `new Date()` instead. Opted to do the minimal change here, but can do the inverse as well. However not sure what would be the implications of doing so.

File: libs/langgraph/package.json (hunk 1 above)
Comment (jacoblee93) [reply libs/langgraph/package.json:20]: I still think we should try to avoid them

File: libs/langgraph/src/pregel/debug.ts (hunk 2 above)
Comment (jacoblee93) [reply libs/langgraph/src/pregel/debug.ts:208]: This seems more natural

**Comment by jacoblee93** [comment]
Looks fine, would just say let's avoid the circular type import. They are cursed in general.

**Comment by georgeevans1995** [comment]
Hi, I'm not sure if this is the right place to report this but the latest release of langgraph studio seems to have broken something. 

We are a team of 6 using langgraph studio all on various version of OS x and all getting the below error. Everything worked ok until 8 hours ago and then docker image langchain/langgraphjs-api updated. This looks like it was also deployed 8 hours ago so thought it may all be connected? 

We are running almost the default setup of this template https://github.com/langchain-ai/retrieval-agent-template-js 

Do you know if there is anything we can do to force lang graph to use a previous docker image? Let me know if there is any other info that would be useful or if there is a better place to report this.

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 60, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/cors.py", line 144, in simple_response
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__
    with collapse_excgroups():
         ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/usr/local/lib/python3.12/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
    raise exc
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/langgraph_license/middleware.py", line 26, in dispatch
    response = await call_next(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/api/langgraph_api/shared/http_logger.py", line 58, in __call__
  File "/api/langgraph_api/shared/http_logger.py", line 52, in __call__
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 62, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 51, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 460, in handle
    await self.app(scope, receive, send)
  File "/api/langgraph_api/auth/middleware.py", line 36, in __call__
  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/api/langgraph_api/shared/route.py", line 118, in handle
  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/api/langgraph_api/shared/route.py", line 37, in app
  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 62, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 51, in wrapped_app
    await app(scope, receive, sender)
  File "/api/langgraph_api/shared/route.py", line 32, in app
  File "/usr/local/lib/python3.12/site-packages/langgraph_storage/retry.py", line 27, in wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/api/langgraph_api/api/threads.py", line 179, in get_thread_history_post
  File "/usr/local/lib/python3.12/site-packages/langgraph_storage/ops.py", line 571, in list
    return [
           ^
  File "/api/langgraph_api/js/remote.py", line 224, in aget_state_history
  File "/api/langgraph_api/js/remote.py", line 180, in _convert_state_snapshot
  File "/api/langgraph_api/js/remote.py", line 156, in _convert_tasks
KeyError: 'path'

**Comment by dqbd** [comment]
Hello @georgeevans1995! Should be fixed now with latest API image, sorry about the issues caused!

**Comment by georgeevans1995** [comment]
Yep thats working again now, thanks for resolving it so quickly

**Pull request merged**